components/normalizer/benches/canonical_composition.rs

```diff
@@ -0,0 +1,46 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use criterion::{BenchmarkId, Criterion};
+use icu_normalizer::properties::CanonicalComposition;
+
+fn function_under_bench(
+    canonical_composer: &CanonicalComposition,
+    composable_points: &[Vec<char>],
+) {
+    composable_points.iter().for_each(|points| {
+        canonical_composer.compose(points[0], points[1]);
+    });
+}
+
+// transform the source part as a vector of characters.
+fn as_vec_char(points: &str) -> Vec<char> {
+    points
+        .split_whitespace()
+        .map(|point| u32::from_str_radix(point, 16).unwrap())
+        .map(char::from_u32)
+        .map(|x| x.unwrap())
+        .collect::<Vec<char>>()
+}
+
+pub fn criterion_benchmark(criterion: &mut Criterion) {
+    let group_name = "canonical_composition";
+    let composer = CanonicalComposition::try_new_unstable(&icu_testdata::unstable()).unwrap();
+
+    let data: Vec<Vec<char>> = include_str!("../tests/data/NormalizationTest.txt")
```

Comment:
The committed data file here is very small and narrow in what it tests (Latin base and accent). I wonder if we'd get more representative data by finding the composites in the test data used for `ComposingNormalizer` and `DecomposingNormalizer` benches and benching those for `CanonicalComposition` and `CanonicalDecomposition`.

Reply:
not using the NormalizationTest.txt, but using bench files and decomposing the character manually.

added the detone for composition benchmark too !